tests/run_tests.py

```diff
@@ -1,32 +1,69 @@
 import argparse
+import importlib
 import os
 import pdb
 import subprocess
 import sys
+import tempfile
 import warnings
 
+from test_utils import __aot_test_cases, print_aot_test_guide
+
 import taichi as ti
 
 
-def _test_cpp():
+def _run_cpp_test(gtest_option="", extra_env=None):
     ti.reset()
     print("Running C++ tests...")
     ti_lib_dir = os.path.join(ti.__path__[0], '_lib', 'runtime')
 
     cpp_test_filename = 'taichi_cpp_tests'
     curr_dir = os.path.dirname(os.path.abspath(__file__))
     build_dir = os.path.join(curr_dir, '../build')
+
     if os.path.exists(os.path.join(build_dir, cpp_test_filename)):
         env_copy = os.environ.copy()
         env_copy['TI_LIB_DIR'] = ti_lib_dir
-        subprocess.check_call(f'./{cpp_test_filename}',
-                              env=env_copy,
-                              cwd=build_dir)
-    else:
-        warnings.warn(
-            f"C++ tests are skipped due to missing {cpp_test_filename} in {build_dir}."
-            "Try building taichi with `TAICHI_CMAKE_ARGS=\'-DTI_BUILD_TESTS:BOOL=ON\' python setup.py develop`"
-            "if you want to enable it.")
+
+        cmd = [f'./{cpp_test_filename}']
+        if gtest_option: cmd.append(gtest_option)
+        if extra_env: env_copy.update(extra_env)
+
+        subprocess.check_call(cmd, env=env_copy, cwd=build_dir)
+
+
+def _test_cpp_aot():
+    tests_visited = []
+    for cpp_test_name, (python_file_path,
+                        function_name) in __aot_test_cases.items():
+        temp_handle = tempfile.TemporaryDirectory()
+        temp_folderpath = temp_handle.name
+
+        # Run AOT Python codes
+        python_file_path = python_file_path.replace("/", ".")
+        if python_file_path.endswith(".py"):
+            python_file_path = python_file_path[:-3]
+        module = importlib.import_module(python_file_path)
+        assert function_name in module.__dict__.keys()
+
+        function_ptr = module.__dict__[function_name]
+        function_ptr(temp_folderpath)
```

Comment:
this part is a bit hacky hmmm

Reply:
It is...Essentially this part is trying to mimic:
```
from cpu_aot.py import aot_compile
aot_compile(folder_path)
```

Do you know if there's any better way of importing a function then execute it? 

The other way around is to fork another subprocess that runs `python3 aot_compile.py --temp_folder=xxx`. However, the forked python process is completely decoupled with the currently python process (the one running run_tests.py started by the user). This could confuse the user due to the diff in environment or even the Python executable (subprocess may not use the same "Python" as the one currently running).

Reply:
@jim19930609 I personally feel `python3 aot_compile.py --temp_folder=xxx` is slightly cleaner and easier to maintain. For example we can use `exec` to execute a file if you want to avoid using a different python. https://stackoverflow.com/questions/1027714/how-to-execute-a-file-within-the-python-interpreter. 
I feel like running a file (so that we can save the mapping from cpp test to python function will simplify things a lot?

Reply:
sounds good, let me refactor this part a little bit.

Reply:
Looks like "exec()" will mess up the Python AST, causing the Taichi compiler to fail. I turned to use "sys.executable" to guarantee Python executable consistency when forking subprocesses.